saleor/graphql/core/mutations.py

```diff
@@ -618,3 +621,31 @@ class Meta:
     @classmethod
     def bulk_action(cls, queryset):
         queryset.delete()
+
+
+class FileUpload(BaseMutation):
+    uploaded_file = graphene.Field(UploadedFile)
+
+    class Arguments:
+        file = Upload(
+            required=True, description="Represents a file in a multipart request."
+        )
+
+    class Meta:
+        description = (
+            "Upload a file. This mutation must be sent as a `multipart` "
+            "request. More detailed specs of the upload format can be found here: "
+            "https://github.com/jaydenseric/graphql-multipart-request-spec"
+        )
+        error_type_class = UploadError
+        error_type_field = "upload_errors"
+
+    @classmethod
+    @staff_member_or_app_required
+    def perform_mutation(cls, _root, info, **data):
+        file_data = info.context.FILES.get(data["file"])
+        path = default_storage.save(file_data._name, file_data.file)
```

Comment:
[django docs about storage save](https://docs.djangoproject.com/en/3.1/ref/files/storage/#django.core.files.storage.Storage.save) states that it returns name of saved file not a path, is it a name or a path?

Reply:
Good point, I was suggesting by [this](https://docs.djangoproject.com/en/3.1/topics/files/#storage-objects), where the returned value is assigned to `path` variable...

Reply:
But, the value returned by saving can be used to open the file from storage, but cannot be open by URL returned by `url` [method](https://docs.djangoproject.com/en/3.1/ref/files/storage/#django.core.files.storage.Storage.url). I'll check it with S3 as storage.

Reply:
I checked with S3 and it works correctly.